scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum export data element description string size"
+version: "1.8"
+name: $T_MAX_PROGRAMMABLE_METRICS_ELEMENT_DESCRIPTION_EXP
+value: "256"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable name string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable description string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable component string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum metric programmable parameter string size"
+version: "1.8"
+name: $T_MAX_METRIC_PROGRAMMABLE_PARAMETER_NAME_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Maximum value information string size"
+version: "1.8"
+name: $T_MAX_VALUE_INFO_CSTRING_EXP
+value: "128"
+--- #--------------------------------------------------------------------------
+type: handle
+desc: "Handle of metric programmable's object"
+version: "1.8"
+class: $tMetricProgrammable
+name: "$t_metric_programmable_exp_handle_t"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Metric Programmable properties queried using $tMetricProgrammableGetPropertiesExp"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_exp_properties_t
+base: $t_base_properties_t
+members:
+    - type: char
+      name: "name[$T_MAX_METRIC_PROGRAMMABLE_NAME_EXP]"
+      desc: "[out] metric programmable name"
+    - type: char
+      name: "description[$T_MAX_METRIC_PROGRAMMABLE_DESCRIPTION_EXP]"
+      desc: "[out] metric programmable description"
+    - type: char
+      name: "component[$T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP]"
+      desc: "[out] metric programmable component"
+    - type: uint32_t
+      name: "tierNumber"
+      desc: "[out] tier number"
+    - type: uint32_t
+      name: "domain"
+      desc: "[out] metric domain number."
+    - type: uint32_t
+      name: "parameterCount"
+      desc: "[out] number of parameters in the programmable"
+    - type: "$t_metric_group_sampling_type_flags_t"
+      name: "samplingType"
+      desc: |
+            [out] metric sampling type.
+            returns a combination of $t_metric_group_sampling_type_flags_t.
+    - type: uint32_t
+      name: "sourceId"
+      desc: "[out] unique metric source identifier(within platform)to identify the HW block where the metric is collected."
```

Comment:
Can the API be extended in the future? If not, consider adding `disaggregation mode` and `query type mask` as new members

Reply:
>Can the API be extended in the future?
Thanks. yes, the api can be extended in the future.

There are a few changes being considered for tracer/event mode of collection.
We could recheck these properties when those changes are integrated as well.